app/cli/cmd/plugins.go

```diff
@@ -0,0 +1,399 @@
+}
+
+func newPluginDescribeCmd() *cobra.Command {
+	var pluginName string
+
+	cmd := &cobra.Command{
+		Use:   "describe",
+		Short: "Show detailed information about a plugin",
+		Args:  cobra.NoArgs,
+		RunE: func(_ *cobra.Command, _ []string) error {
+			if pluginName == "" {
+				return fmt.Errorf("plugin name is required")
+			}
+
+			result, err := action.NewPluginDescribe(actionOpts, pluginManager).Run(context.Background(), pluginName)
+			if err != nil {
+				return err
+			}
+
+			if flagOutputFormat == formatJSON {
+				type pluginDetail struct {
+					Name        string                `json:"name"`
+					Version     string                `json:"version"`
+					Description string                `json:"description"`
+					Path        string                `json:"path"`
+					Commands    []plugins.CommandInfo `json:"commands"`
+				}
+
+				detail := pluginDetail{
+					Name:        result.Plugin.Metadata.Name,
+					Version:     result.Plugin.Metadata.Version,
+					Description: result.Plugin.Metadata.Description,
+					Path:        result.Plugin.Path,
+					Commands:    result.Plugin.Metadata.Commands,
+				}
+
+				return encodeJSON(detail)
+			}
+
+			pluginInfoTableOutput(result.Plugin)
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&pluginName, "name", "", "", "Name of the plugin to describe (required)")
+	cobra.CheckErr(cmd.MarkFlagRequired("name"))
+
+	return cmd
+}
+
+func newPluginDownloadCmd() *cobra.Command {
+	var file string
+	var filename string
+	var useAlternate bool
+	var accessKeyID string
+	var secretAccessKey string
+	var location string
+	var region string
+
+	cmd := &cobra.Command{
+		Use:   "download",
+		Short: "Download a plugin from S3 or a HTTP URL",
+		Long:  "Download a plugin from a specified URL or S3 bucket to the plugins directory. By default, uses S3 download unless --alternate flag is specified.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+
+			opts := &action.PluginDownloadOptions{
+				File:            file,
+				Filename:        filename,
+				UseAlternate:    useAlternate,
+				AccessKeyID:     accessKeyID,
+				SecretAccessKey: secretAccessKey,
+				Location:        location,
+				Region:          region,
+			}
+
+			result, err := action.NewPluginDownload(actionOpts, pluginManager).Run(ctx, opts)
+			if err != nil {
+				return fmt.Errorf("failed to download plugin: %w", err)
+			}
+
+			fmt.Printf("Plugin downloaded successfully to: %s\n", result.FilePath)
+			return nil
+		},
+	}
+
+	// Common flags
+	cmd.Flags().StringVarP(&file, "file", "f", "", "S3 object key of the plugin to download or a URL of the file (required)")
+	cmd.Flags().StringVarP(&filename, "filename", "", "", "Custom filename to save the plugin as (optional)")
+	cmd.Flags().BoolVar(&useAlternate, "alternate", false, "Use HTTP download instead of S3 (optional)")
+
+	// S3 specific flags
+	cmd.Flags().StringVar(&accessKeyID, "access-key-id", "", "AWS Access Key ID for S3 download")
```

Comment:
I didn't know that s3 was in the scope of this feature, I was thinking to just support HTTP download, like we do in other commands.

I personally wouldn't implement s3 this way either, I'd not use credentials like this and instead leverage s3 credentials handling, which will allow the user to authenticate using .credentials files, env vars, etc out of the box

That said, I'd not even add support to S3 in this PR. WDYT?